src/package.c

```diff
@@ -975,18 +975,70 @@ int apk_ipkg_add_script(struct apk_installed_package *ipkg,
 	return 0;
 }
 
-void apk_ipkg_run_script(struct apk_installed_package *ipkg,
-			 struct apk_database *db,
-			 unsigned int type, char **argv)
+static int run_script(char** argv, int dirfd, char* fn)
 {
+	int status;
+	pid_t pid;
 	static char * const environment[] = {
 		"PATH=/usr/sbin:/usr/bin:/sbin:/bin",
 		NULL
 	};
+
+	pid = fork();
+	if (pid == -1) {
+		apk_error("fork() failed with error str: %s", apk_error_str(errno));
+		return -1;
+	}
+	if (pid == 0) {
+		umask(0022);
+		if (fchdir(dirfd) == 0) {
+			if (fn == NULL)
+				execve(argv[0], argv, environment); /* pre/post hook */
```

Comment:
yes that a bit tricky... would preventing this type of hooks when running in an alternate root be a way forward you think?